x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushTopNToSource.java

```diff
@@ -60,7 +62,9 @@
 public class PushTopNToSource extends PhysicalOptimizerRules.ParameterizedOptimizerRule<TopNExec, LocalPhysicalOptimizerContext> {
     @Override
     protected PhysicalPlan rule(TopNExec topNExec, LocalPhysicalOptimizerContext ctx) {
-        Pushable pushable = evaluatePushable(topNExec, x -> LucenePushDownUtils.hasIdenticalDelegate(x, ctx.searchStats()));
+        Predicate<FieldAttribute> hasIdenticalDelegate = (fa) -> LucenePushDownUtils.hasIdenticalDelegate(fa, ctx.searchStats());
```

Comment:
would be useful to rename this method to something more meaningful such as `hasExactFallback`.

Reply:
While this method pre-dates my work (was added almost a year ago), I agree that the name is not entirely clear, and my last commit also changed slightly what is checked deeper down too. However, the name `hasExactFallback` is also insufficient (and to my ears sounds synonymous with what we already have). We check if the TEXT field-type has a surrogate and then make several checks on the nature of that surrogate. Any name that fully captures what is checked will be unreasonably long, which is why I kept the original name coded before.